doc/scenario.md

```diff
@@ -0,0 +1,392 @@
+
+Input facts (set by role):
+- `instance_info` - information for a current instance ([more details here](#role-facts-descriptions)).
+
+Input facts (set by config):
+- `cartridge_multiversion` - indicates that multiple package versions is enabled;
+- `cartridge_app_user` - user which will own the links;
+- `cartridge_app_group` - group which will own the links.
+
+### configure_instance
+
+Configure instance in runtime and change instance config.
+
+Input facts (set by role):
+- `instance_info` - information for a current instance ([more details here](#role-facts-descriptions)).
+
+Input facts (set by config):
+- `config` - instance configuration ([more details here](/README.md#instances));
+- `restarted` - if instance should be restarted or not (user forced decision);
+- `expelled` - indicates if instance must be expelled from topology;
+- `stateboard` - indicates that the instance is a stateboard;
+- `cartridge_app_name` - application name;
+- `cartridge_cluster_cookie` - cluster cookie for all cluster instances;
+- `cartridge_defaults` - default configuration parameters values for instancesж
+- `cartridge_app_user` - user which will own the links;
+- `cartridge_app_group` - group which will own the links.
+
+Output facts:
+- `needs_restart` - if instance should be restarted to apply code or configuration changes.
+
+### restart_instance
+
+Restart instance if it should be restarted.
+
+Input facts (set by role):
+- `needs_restart` - if instance should be restarted to apply code or configuration changes;
+- `instance_info` - information for a current instance ([more details here](#role-facts-descriptions)).
+
+Input facts (set by config):
+- `restarted` - if instance should be restarted or not (user forced decision).
+
+### wait_instance_started
+
+Wait until instance is fully started.
+
+Input facts (set by role):
+- `instance_info` - information for a current instance ([more details here](#role-facts-descriptions)).
+
+Input facts (set by config):
+- `expelled` - indicates if instance must be expelled from topology;
+- `stateboard` - indicates that the instance is a stateboard;
+- `instance_start_timeout` - time in seconds to wait for instance to be started;
+- `cartridge_wait_buckets_discovery` - indicates if routers should wait for buckets discovery after vshard bootstrap.
+
+### connect_to_membership
+
+Connect instance to membership.
+
+Input facts (set by role):
+- `instance_info` - information for a current instance ([more details here](#role-facts-descriptions)).
+
+Input facts (set by config):
+- `expelled` - indicates if instance must be expelled from topology;
+- `stateboard` - indicates that the instance is a stateboard;
+- `cartridge_app_name` - application name;
+- `config` - instance configuration ([more details here](/README.md#instances)).
+
+### set_control_instance
+
+Find some instance that is already in cluster. If there is no such instance, use one of instances that should be joined.
```

Comment:
Should we notice that this task makes sense only after running `connect_to_membership`?